### PR TITLE
Handling More Http Methods

### PR DESCRIPTION
**Pull request by mikaelm12**
@Tratcher @davidfowl  @victorhurdugaci

### CODE REVIEW

File: src/Microsoft.AspNet.Proxy/ProxyMiddleware.cs [hunk 1]
```diff
@@ -58,7 +58,10 @@ public ProxyMiddleware([NotNull] RequestDelegate next, [NotNull] ProxyOptions op
         public async Task Invoke(HttpContext context)
         {
             var requestMessage = new HttpRequestMessage();
-            if (string.Equals(context.Request.Method, "POST", StringComparison.OrdinalIgnoreCase))
+            if (!string.Equals(context.Request.Method, "GET", StringComparison.OrdinalIgnoreCase) ||
```
Comment (Tratcher) [inline src/Microsoft.AspNet.Proxy/ProxyMiddleware.cs]: Tests?
Shouldn't this be `&&` instead of `||`?

**Comment by mikaelm12** [comment]
Updated but I'm also going to switch to using theory tests to clean it up a bit

**Comment by mikaelm12** [comment]
Updated

**Comment by Tratcher** [comment]
:shipit:

**Pull request merged**